osu.Game/Tests/Visual/OsuTestScene.cs

```diff
@@ -203,7 +203,7 @@ protected APIBeatmapSet CreateAPIBeatmapSet(RulesetInfo ruleset = null)
 
             return new APIBeatmapSet
             {
-                OnlineID = beatmap.BeatmapSet.OnlineID,
+                OnlineID = ((IBeatmapSetInfo)beatmap.BeatmapSet).OnlineID,
```

Comment:
This one's scary because the implications don't show up in the diffs of this PR, but I'll trust that you've gone through every prior usage of `OnlineID` which turned `(int)-1` for a `null` value don't suddenly break because they now return the straight `null` value.

Reply:
I'm a bit confused. The target here is an API type and the source is a `BeatmapSetInfo`. It was already doing the propagation using the interface type, which I've kept by using the case, no?

Reply:
You're correct, but if there are any other cases doing the same thing, they won't be shown in the PR. So I have to trust that there are no such other cases.

Reply:
The dangerous thing is that now the nullability of the return type depends on which type is being handled.

The good thing is that because the prior type was a value type, you can't just replace `int` to `int?` and have it compile. So any mis-usages would presumably not compile (and that's regardless of NRT). But I also find this a little spooky.

Reply:
> you can't just replace int to int? and have it compile

For the most parts, yes, but my biggest concern is around `(int) == (int?)` which I believe is a valid operation that this could change the semantics of.

E.g. `if (beatmap.OnlineID == -1)` is perfectly valid for both `int` and `int?` so the semantics here could change.

Reply:
The changes were made using rider refactors so I'm 99% sure it's safe in this case, but I understand the concerns now.